### DISCUSSION TITLE
Provisioner \runner-images\images\win/scripts/Installers/Wait-WindowsUpdatesForInstall.ps1 hangs at KB5031221

### DISCUSSION THREAD

**Discussion by daniatic** [actions/virtual-environments#8788] [General questions]
When running the windows2022 image the provisioner \runner-images\images\win/scripts/Installers/Wait-WindowsUpdatesForInstall.ps1 is waiting for the update "2023-10 Cumulative Update for .NET Framework 3.5, 4.8 and 4.8.1 for Microsoft server operating system version 21H2 for x64 (KB5031221)" to complete, until all retry counts are used up and fails in the end. I also increased the retries and the retry interval to see if it resolves after enough time, but with no success. 

As I removed some provisioners (in order to reduce the image to my needs) I do not know if that is the cause or if there is a general problem. So before I create a bug, I wanted to ask here for some advice.

**Comment by daniatic** [comment]
I raised the retry interval and the retry count, so that it waited for more than half an hour. Even with a stronger machine (F16s_v2) and the extended retry it failed. 

When I tried it yesterday it mostly failed but once when it was successful with 5 retries left.

**Reply by Alexey-Ayupov** [reply]
We generate images every night and have not faced such a problem. It is hard to say what's the root cause since we do not know what exactly was removed from the image on your side. It also could be network problems on your or Azure's side.

**Reply by daniatic** [reply]
I just removed some of the tools, like Java, Ruby, python and so forth. I'd assume that Windows Update would succeed, because it identifies on itw own, which updates to install, without me telling it to do so. 
Regarding network problems I don't think that's the case. The runner was able to download other stuff from different sources and also installed several other windows updates before failing with the mentioned one. 

As a temporary solution I disabled KB5031221 within Install-WindowsUpdate.ps1 and now two of the just released November Updates don't go through: KB5032198, KB5032336

Maybe it has something to do with my setting but I'm not totally convinced, also partially due to a past experience. 

The last time I had trouble, it was with KB5001148, when I was desperatly looking for a solution, only to later find a new commit which excluded this from windows update.

**Reply by FiyaasAhmedKhan** [reply]
> I raised the retry interval and the retry count, so that it waited for more than half an hour. Even with a stronger machine (F16s_v2) and the extended retry it failed. 
> 
> When I tried it yesterday it mostly failed but once when it was successful with 5 retries left.

**Reply by FiyaasAhmedKhan** [reply]
> > I raised the retry interval and the retry count, so that it waited for more than half an hour. Even with a stronger machine (F16s_v2) and the extended retry it failed. 
> > 
> > When I tried it yesterday it mostly failed but once when it was successful with 5 retries left. 
> 
>

**Comment by FiyaasAhmedKhan** [comment]
> When running the windows2022 image the provisioner \runner-images\images\win/scripts/Installers/Wait-WindowsUpdatesForInstall.ps1 is waiting for the update "2023-10 Cumulative Update for .NET Framework 3.5, 4.8 and 4.8.1 for Microsoft server operating system version 21H2 for x64 (KB5031221)" to complete, until all retry counts are used up and fails in the end. I also increased the retries and the retry interval to see if it resolves after enough time, but with no success.
> 
> As I removed some provisioners (in order to reduce the image to my needs) I do not know if that is the cause or if there is a general problem. So before I create a bug, I wanted to ask here for some advice.
